subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/PmdInvoker.groovy

```diff
@@ -131,6 +131,15 @@ class PmdInvoker implements Action<AntBuilderDelegate> {
             }
 
             reports.each { report ->
+                if (report.name.get() == "sarif" && version < VersionNumber.parse("6.31.0")) {
+                    // https://github.com/pmd/pmd/releases/tag/pmd_releases%2F6.31.0
+                    return
+                }
+                else if (report.name.get() == "codeclimate" && version < VersionNumber.parse("5.3.7")) {
+                    // https://github.com/pmd/pmd/releases/tag/pmd_releases%2F5.3.7
+                    return
+                }
```

Comment:
If we would disable these new report types by default I guess we could possibly skip these checks too :thinking: